pkg/traceql/ast.go

```diff
@@ -23,7 +33,8 @@ type typedExpression interface {
 }
 
 type RootExpr struct {
-	Pipeline Pipeline
+	Pipeline        Pipeline
+	MetricsPipeline metricsFirstStageElement
```

Comment:
Having the metrics pipeline split from the spanset filter pipeline isn't ideal, but best for now.   We send the eval callback of the spanset pipeline into the storage SecondPass, and the metrics pipeline has a different signature (output is time-series).  Long-term this break needs to be closed and end up with one pipeline, but this current form will carry the functionality a long way.  This will work until we add cross time-series arithmetic like: 

```
  ({a} | rate()) / 
  ({b} | rate())
```